hal/halx86/generic/x86bios.c

```diff
@@ -295,7 +296,18 @@ HalInitializeBios(
         DPRINT1("memory: %p, %p\n", *(PVOID*)x86BiosMemoryMapping, *(PVOID*)(x86BiosMemoryMapping + 8));
         //DbgDumpPage(x86BiosMemoryMapping, 0xc351);
 
-        x86BiosIsInitialized = TRUE;
+        /* Initialize the emulator context */
```

Comment:
What's the point of this? You are actively introducing a limitation.

Reply:
1. Why would be the point of having concurrent INT10h (or other BIOS) calls?
2. Reinitializing the emulator for each INT10h call is not very efficient.

Reply:
1. I don't know. But that is not an argument to change it.
2. "not very efficient" - OK, so how many cycles are we talking about? And how often is this called? And how many cycles does the whole thing cost? And most importantly: what is the impact of having to access likely uncached memory (the global data) vs accessing likely cached memory (the stack) on access time and cache/TLB usage? You probably measured it, right? Otherwise I'd call it premature optimization, and you know what that is.

Reply:
My 2 cents:

From a code point of view:
* `Fast486Initialize()` each time : safest.
* `Fast486Initialize()` once, then `Fast486Reset()` each time : a bit less safe only, assumes fixed init values never get changed anyhow.
* `Fast486Initialize()` once, and only : less safe, assumes first `Fast486Reset()` is enough.

As for speed/cache, I don't know, but notice new code is accessing other globals (initialized, count, ...) anyway.